weight.go

```diff
@@ -0,0 +1,53 @@
+package netmap
+
+type (
+	// AggregatorFactory is a Factory for a specific Aggregator
+	AggregatorFactory struct {
+		New func() Aggregator
+	}
```

Comment:
There are two approaches of using `Aggregator` in this PR:
1) Using `Aggregator` directly in `Traverse` function
```go
func (b *Bucket) Traverse(a Aggregator, wf WeightFunc) Aggregator
```
2) Using `AggregatorFactory` in `TraverseTree` function
```go
func (b *Bucket) TraverseTree(af AggregatorFactory, wf WeightFunc)
```
Can you elaborate what provides factory pattern in the second case? Can we use single approach in both functions?

Reply:
When a weight of is computed, we need to be able to instantiate some kind of an aggregator for every bucket. Therefore we need some way to create `Aggregator`s. This is done via `AggregatorFactory`.
I have also considered to use `Clear()` method of `Aggregator`, but found it dirty (more possible bugs because of memory reuse, no ability to parallelize work).

Reply:
I see. `Traverse()` is used for every single bucket and it requires fresh `Aggregator` instance. And `TraverseTree()` is used for a whole graph and it produces `Aggregator`s with `AggregatorFactory`. Thank you.